packages/astro/src/runtime/server/hydration.ts

```diff
@@ -0,0 +1,125 @@
+import type { AstroComponentMetadata } from '../../@types/astro-core';
+import type { SSRElement } from '../../@types/astro-runtime';
+import { valueToEstree } from 'estree-util-value-to-estree';
+import * as astring from 'astring';
+import { serializeListValue } from './util.js';
+
+const { generate, GENERATOR } = astring;
+
+// INVESTIGATE: What features are we getting from this that we need?
+// JSON.stringify has a "replacer" argument.
+// A more robust version alternative to `JSON.stringify` that can handle most values
+// see https://github.com/remcohaszing/estree-util-value-to-estree#readme
+const customGenerator: astring.Generator = {
+  ...GENERATOR,
+  Literal(node, state) {
+    if (node.raw != null) {
+      // escape closing script tags in strings so browsers wouldn't interpret them as
+      // closing the actual end tag in HTML
+      state.write(node.raw.replace('</script>', '<\\/script>'));
+    } else {
+      GENERATOR.Literal(node, state);
+    }
+  },
+};
+
+// Serializes props passed into a component so that they can be reused during hydration.
+// The value is any 
```

Comment:
Not a request for change, but my only suggestion here is that we are consistent with commenting style, and we enforce with linting. Most of the codebase has JSDoc-style comments. If we want to do only single-line comments like Fred likes, I’m fine with that too! Could not care less! But I just want consistency (I’m a big fan of a sweeping PR to just change all at once because it’s not hard)